pkg/controllers/user/pipeline/controller/pipeline/pipeline.go

```diff
@@ -49,7 +49,9 @@ func (l *Lifecycle) Updated(obj *v3.Pipeline) (runtime.Object, error) {
 func (l *Lifecycle) Remove(obj *v3.Pipeline) (runtime.Object, error) {
 	if obj.Status.WebHookID != "" {
 		if err := l.deleteHook(obj); err != nil {
-			return obj, err
+			// attempt to clean up the hook but do not pend on it on exceptions.
+			logrus.Warnf("Fail to delete hook: %v", err)
```

Comment:
Make this logic speicifc to NotFound errors. That's a common pattern we use: attempt to delete. If error is a "not found" error, then return successfully. No need to even log a warning in this case.

If there is any other error, return it.

Reply:
Updated